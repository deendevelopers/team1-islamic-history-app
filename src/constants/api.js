export const API_BASE_URL = "http://localhost:3004";
export const API_HISTORIC_EVENTS_URL = API_BASE_URL + "/events";