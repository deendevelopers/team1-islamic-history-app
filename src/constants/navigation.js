export const AUTH_STACK = "AUTH_STACK";
export const APP_STACK = "APP_STACK";

export const SPLASH_SCREEN = "SPLASH_VIEW";
export const LOGIN_VIEW = "LOGIN_VIEW";
export const HOME_VIEW = "HOME_VIEW";

export const LIKES_VIEW = "LIKES_VIEW";
export const ACCOUNTS_VIEW = "ACCOUNTS_VIEW";
export const VIRO_VIEW = "VIRO_VIEW";

export const SEARCH_VIEW = "SEARCH_VIEW";
export const CARDS_PREVIEW_VIEW = "CARDS_PREVIEW_VIEW";
export const CARDS_SELECTED_VIEW = "CARDS_SELECTED_VIEW";
export const AUTH_VIEW = "Auth";
export const APP_VIEW = "App";



export const FACT_VIEW = "FACTS_VIEW";