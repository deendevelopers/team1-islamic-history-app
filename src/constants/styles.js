export const FACT_COLOR = "#d1cfce";