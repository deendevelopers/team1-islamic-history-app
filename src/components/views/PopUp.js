import React from 'react';
import PropTypes from 'prop-types';
import styled, {css} from 'styled-components'
import {BORDER_DEFAULT_COLOR, VIEW_BACKGROUND_COLOR} from "../../constants/styles";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {SwipeUpEntryKeyFrame, SwipeUpExitKeyFrame} from "../keyframes/SwipeUpKeyFrame";
import {POPUP_ANIMATION_TIME_MILLIS} from "../../constants/animation";

const PopUpPanel = styled.div`
  height: 100%;
  width: 100%;
  position: absolute;
  justify-content: center;
  align-items: flex-end;
  display: flex;
  z-index: 10;
  overflow: hidden;
  animation-name: ${props => props.hide ? SwipeUpExitKeyFrame : SwipeUpEntryKeyFrame};
  animation-duration: ${POPUP_ANIMATION_TIME_MILLIS}ms;
  animation-timing-function: ease-in-out;
  animation-fill-mode: both;
  animation-direction: normal;
  //animation-iteration-count: infinite;
`;
PopUpPanel.displayName = "PopUpPanel";

const PopUpDiv = styled.div`
  background-color: #FFFFFF;
  min-height: 20%;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  z-index: 100;
  ${props => props.fullscreen && css`
    height: 100%;
    width: 100%;
    justify-content: start !important;
    padding-bottom: 0 !important;
  `}
`;
PopUpDiv.displayName = "PopUpDiv";

const TitleDiv = styled.div`
  position: relative;
  border-bottom: 1px solid ${BORDER_DEFAULT_COLOR};
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 6vh;
`;

const Heading = styled.h2`
  font-weight: bold;
  margin: 0;
  line-height: inherit;
  text-transform: uppercase;
  //font-size: 1.25rem;
  font-family: inherit;
`;
Heading.displayName = "Heading";

const CloseButton = styled.div`
  position: absolute;
  right: 0;
  background: none;
  height: 100%;
  width: 50px;
  text-align: center;
  align-items: center;
  justify-content: center;
  display: flex;
`;
CloseButton.displayName = "CloseButton";


export default class PopUp extends React.Component {

  static propTypes = {
    title: PropTypes.string.isRequired,
    hidePopUpPanel: PropTypes.func.isRequired,
    fullscreen: PropTypes.bool,
  };

  constructor(props) {
    super(props);

    this.state = {
      hidden: false
    };

    this.hide = this.hide.bind(this);
  }

  hide() {
    this.setState({hidden: true}, () => {
      setTimeout(() => {
        this.props.hidePopUpPanel();
      }, POPUP_ANIMATION_TIME_MILLIS)
    });
  }

  render() {
    return (
        <PopUpPanel hide={this.state.hidden} onClick={this.hide}>
          <PopUpDiv fullscreen={this.props.fullscreen} onClick={(event) => event.stopPropagation()}>
            <TitleDiv>
              <CloseButton onClick={this.hide}>
                <FontAwesomeIcon icon={"times"}/>
              </CloseButton>
              <Heading>{this.props.title}</Heading>
            </TitleDiv>
            {this.props.children}
          </PopUpDiv>
        </PopUpPanel>
    )
  }
}