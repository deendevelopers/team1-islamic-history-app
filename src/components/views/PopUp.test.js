import React from 'react';
import renderer from 'react-test-renderer';
import {cleanup} from '@testing-library/react';
import {mount, shallow} from 'enzyme';
import PopUp from "./PopUp";
import {POPUP_ANIMATION_TIME_MILLIS} from "../../constants/animation";

describe("PopUp", () => {

  const dummyFunction = () => {
  };

  jest.useFakeTimers();

  afterEach(cleanup);

  it('renders correctly', () => {
    const popUp = renderer.create(
        <PopUp
            title={"some title"}
            hidePopUpPanel={dummyFunction}
            fullscreen={false}
        >
          <div>children</div>
        </PopUp>);
    expect(popUp).toMatchSnapshot();
  });

  it('should close when pop up panel is clicked', function () {
    const hidePopUpPanel = jest.fn();
    const popUp = shallow(
        <PopUp
            title={"title"}
            hidePopUpPanel={hidePopUpPanel}
            fullscreen={false}
        />);

    popUp.find("PopUpPanel").simulate("click");
    jest.runAllTimers();

    expect(popUp.state("hidden")).toBeTruthy();
    expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), POPUP_ANIMATION_TIME_MILLIS);
    expect(hidePopUpPanel).toHaveBeenCalled();
  });

  it("should stop propagation on click", () => {
    const hidePopUpPanel = jest.fn();
    const mockEvent = jest.fn();
    const popUp = shallow(
        <PopUp
            title={"title"}
            hidePopUpPanel={hidePopUpPanel}
            fullscreen={false}
        />);

    popUp.find("PopUpDiv").simulate("click", {stopPropagation: mockEvent});

    expect(mockEvent).toHaveBeenCalled();
    expect(hidePopUpPanel).not.toHaveBeenCalled();
  });

  it('should call hidePopUpPanel on close click', () => {
    const hidePopUpPanel = jest.fn();
    const popUp = mount(
        <PopUp
            title={"title"}
            hidePopUpPanel={hidePopUpPanel}
            fullscreen={false}
        />);

    popUp.find("CloseButton").first().simulate("click");
    jest.runAllTimers();

    expect(hidePopUpPanel).toHaveBeenCalled();
  });

});
