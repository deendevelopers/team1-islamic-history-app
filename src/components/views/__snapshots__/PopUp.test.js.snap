// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUp renders correctly 1`] = `
<div
  className="PopUp__PopUpPanel-sc-1pjswbo-0 jVzsf"
  onClick={[Function]}
>
  <div
    className="PopUp__PopUpDiv-sc-1pjswbo-1 SYVVD"
    onClick={[Function]}
  >
    <div
      className="PopUp__TitleDiv-sc-1pjswbo-2 jRSClx"
    >
      <div
        className="PopUp__CloseButton-sc-1pjswbo-4 daVERV"
        onClick={[Function]}
      />
      <h2
        className="PopUp__Heading-sc-1pjswbo-3 jeSsEs"
      >
        some title
      </h2>
    </div>
    <div>
      children
    </div>
  </div>
</div>
`;
